src/Infrastructure.EntityFramework/Repositories/Queries/UserCipherDetailsQuery.cs

```diff
@@ -8,11 +8,21 @@ namespace Bit.Infrastructure.EntityFramework.Repositories.Queries;
 public class UserCipherDetailsQuery : IQuery<CipherDetails>
 {
     private readonly Guid? _userId;
-    public UserCipherDetailsQuery(Guid? userId)
+    private readonly bool _useFlexibleCollections;
+    public UserCipherDetailsQuery(Guid? userId, bool useFlexibleCollections)
     {
         _userId = userId;
+        _useFlexibleCollections = useFlexibleCollections;
     }
+
     public virtual IQueryable<CipherDetails> Run(DatabaseContext dbContext)
+    {
+        return _useFlexibleCollections
+            ? Run_V2(dbContext)
+            : Run_Legacy(dbContext);
+    }
+
+    private IQueryable<CipherDetails> Run_Legacy(DatabaseContext dbContext)
```

Comment:
⛏️ The "legacy" seems unnecessary to me as it's the V1. With thorough cleanup this returns and the "V2" is dropped anyway. Per some past conversations this is why I like tech debt tasks for cleanup and restoring naming, so this could be "vNext" even. Don't need to dwell too much on naming though.

Reply:
I'm warming to the vNext idea, just so we don't have ever-incrementing version numbers.

Reply:
Changed to `VNext` and `VCurrent` (??) or would you go back the legacy wording in this pattern?

Reply:
`Run` and `Run_vNext` feel fine to me with the consideration that this is a transitional state.

Reply:
`Run` is already taken as the public interface that just switches between the 2 private methods.